proposals/async-streams.md

```diff
@@ -72,7 +72,7 @@ try
         Use(enumerator.Current);
     }
 }
-finally { await enumerator.DisposeAsync(); }
+finally { await ((IAsyncDisposable)enumerator).DisposeAsync(); }
```

Comment:
Cast is unnecessary since `IAsyncDisposable<out T>` implements `IAsyncDisposable`.

Reply:
Fair, not true for a pattern based implementation though. I'll remove.